### PR TITLE
Added check mode for all modules

### PR DESCRIPTION
**Pull request by SebastianGode**
Currently just half of all Modules
WIP

### CODE REVIEW

**Comment by SebastianGode** [comment]
Adressing Issue https://github.com/opentelekomcloud/ansible-collection-cloud/issues/86

**Comment by SebastianGode** [comment]
All modules should have check_mode support now. 

I tested with the existing Integration Tests and just added `check_mode: true` where necessary. 


Info Modules generally don't change anything so check mode doesn't do anything there. Except if I misunderstood that part. As a result I just added 
```
module_kwargs = dict(
     supports_check_mode=True
)
```
to the Info Modules without changing anything else. The already existing availability_zone_info module also hasn't changed anything else (https://github.com/opentelekomcloud/ansible-collection-cloud/blob/master/plugins/modules/availability_zone_info.py) so hopefully that's enough.

**Review by gtema — CHANGES_REQUESTED** [review]
Generally very good, but please also change existing integration tests that they verify this (see https://github.com/opentelekomcloud/ansible-collection-cloud/blob/master/tests/integration/targets/waf_domain/tasks/main.yaml#L97 for example)

**Comment by SebastianGode** [comment]
@gtema I added integration tests for check_mode on modules where it makes sense (so no Info modules as it won't change anything).

**Review by tischrei — APPROVED** [review]

**Review by gtema — APPROVED** [review]

File: tests/integration/targets/ces/tasks/main.yaml [hunk 1]
```diff
@@ -149,4 +177,4 @@
           alarm_name: "{{ alarm_name }}"
           state: absent
         register: dns_rs_dr
-      ignore_errors: yes
+      ignore_errors: yes
```
Comment (anton-sidelnikov) [inline tests/integration/targets/ces/tasks/main.yaml:180]: empty line

File: tests/integration/targets/dns/tasks/main.yaml [hunk 2]
```diff
@@ -246,4 +368,4 @@
           name: "{{ network_name }}"
           state: absent
         register: dns_net_dr
-      ignore_errors: yes
+      ignore_errors: yes
```
Comment (anton-sidelnikov) [inline tests/integration/targets/dns/tasks/main.yaml:371]: empty line

File: tests/integration/targets/nat/tasks/main.yaml [hunk 3]
```diff
@@ -222,4 +280,4 @@
           name: "{{ network_name }}"
           state: absent
         register: nat_net
-      ignore_errors: yes
+      ignore_errors: yes
```
Comment (anton-sidelnikov) [inline tests/integration/targets/nat/tasks/main.yaml:283]: empty line

**Review by anton-sidelnikov — APPROVED** [review]

**Pull request merged**